docs/internals/optimizer.rst

```diff
@@ -1096,6 +1097,48 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and ``mstore`` statements if conditions are met for said removal.
+In case of an ``sstore``, if all outgoing code paths lead to a revert or to another ``sstore`` for which the optimizer
```

Comment:
This isn't mentioning an explicit `revert()` though, it's just saying revert, whatever the cause of said revert may be is kind of irrelevant here, no?

Reply:
I know, but it could be taken that way. I think it's not obvious that the optimizer would actually detect infinite recursion and treat it as a revert so I think it's worth mentioning.

Reply:
I disagree with this, as I think we'd have to go through the entire optimizer doc and mention infinite recursion detection and the subsequent `revert` due to it in every instance; I feel as though it's too much detail for something that should give a broad explanation of an optimizer step. @ekpyron @chriseth, any of you want to be the tie breaker?

Reply:
I mean, it doesn't say ``revert``, but revert :-D...
But yeah, if we want to go with @nikola-matic's argumentation I'd rather say ``if all outgoing code paths revert`` not ``lead to a revert``. 

But yeah, it then doesn't hurt to make it

```
In case of an ``sstore``, if all outgoing code paths revert (due to an explicit ``revert()`` or infinite recursion) or lead to another ``sstore`` for which the optimizer
```

As for checking the other optimizer steps: we actually *should* do that :-). In ``DeadCodeEliminator`` the recursion case is, for example, also missing (and there ``revert`` is used in a list of statements, so clearly refers to the builtin call only)...

Reply:
On the other hand I also think the ``DeadCodeEliminator`` is also the *only* case that needs adjustment, so it's not too bad.